libtiny_client/src/state.rs

```diff
@@ -109,13 +117,76 @@ struct StateInner {
     server_info: ServerInfo,
 }
 
+#[derive(Debug)]
+struct Chan {
+    /// Name of the channel
+    name: String,
+    /// Set of nicknames in channel
+    nicks: HashSet<String>,
+    /// Channel joined state
+    join_state: JoinState,
+    /// Join attempts
+    join_attempts: u8,
```

Comment:
Moved join_attempts out of the Joining variant in JoinState because when the state is Initial there should be join attempts. I thought it's nicer than keeping the value in Initial and in Joining.